### PR TITLE
windows: add tty unicode support for input

### PR DESCRIPTION
**Pull request by retep998**
Switches from using ReadConsoleA to ReadConsoleW.
Uses an auxiliary buffer to store the temporary utf-16 string.

### CODE REVIEW

File: src/win/tty.c [hunk 1]
```diff
@@ -303,6 +303,8 @@ static DWORD CALLBACK uv_tty_line_read_thread(void* data) {
   uv_tty_t* handle;
   uv_req_t* req;
   DWORD bytes, read_bytes;
+  WCHAR utf16[8192 / 3];
```
Comment (saghul) [inline src/win/tty.c]: Since this 8192 si related to line 320, can you please add a define with it?

Comment (retep998) [inline src/win/tty.c]: Added MAX_INPUT_BUFFER_LENGTH

**Comment by saghul** [comment]
LGTM. Can either of you guys have a quick look? @piscisaureus @orangemocha. Thanks!

**Comment by piscisaureus** [comment]
lgtm too, ship it

**Comment by saghul** [comment]
Thanks a lot @retep998, landed in https://github.com/joyent/libuv/commit/68cd6d6a6ff7b4088b210498f81002080733fe5f

**Pull request closed without merging**